crates/turbopack-ecmascript/src/references/esm/base.rs

```diff
@@ -34,17 +34,17 @@ use crate::{
 #[turbo_tasks::value]
 pub enum ReferencedAsset {
     Some(Vc<Box<dyn EcmascriptChunkPlaceable>>),
-    OriginalReferenceTypeExternal(String),
+    External(String, ExternalType),
     None,
 }
 
 impl ReferencedAsset {
     pub async fn get_ident(&self) -> Result<Option<String>> {
         Ok(match self {
             ReferencedAsset::Some(asset) => Some(Self::get_ident_from_placeable(asset).await?),
-            ReferencedAsset::OriginalReferenceTypeExternal(request) => {
-                Some(magic_identifier::mangle(&format!("external {}", request)))
-            }
+            ReferencedAsset::External(request, ty) => Some(magic_identifier::mangle(&format!(
+                "{ty} external {request}"
```

Comment:
This changes the ident for externals to include the type. We need to include it to avoid conflicts between multiple externals that only differ in type.